### DISCUSSION TITLE
Handeling migrations with Prisma?

### DISCUSSION THREAD

**Discussion by enricoschaaf** [supabase/supabase#554] [Questions] [answered]
It it it good idea to use the Prisma CLI as a tool to handle migrations or could I potentially ruin stuff thats setup my superbase?
If its not does anyone got any recommendations for diffrent tools?

**Comment by shorn** [comment]
I use a java tool called flyway in my demo app: https://github.com/kopi-cloud/cabbage/tree/main/database

Likely not appropriate for you unless you're already comfortable in a JVM environment.

One thing i can tell you is to be careful of any "clean schema"  functionality, see the linked readme.

**Comment by kiwicopple** [comment, marked as answer]
@enricoschaaf - it should be fine as long as it's only updating the `public` schema, and it doesn't remove any of the roles.

We also have an `auth` schema and an `extensions` schema, but otherwise supabase is "just" Postgres. Let us know if you spot any problems errors - YMMV since I haven't used the prisma migration tool personally

**Reply by enricoschaaf** [reply]
I tested it and auth and extensions seem to work just fine, will report if I experience any errors.

**Reply by enricoschaaf** [reply]
I ended up not using it further because:
1. It alters my column data types because it tries to conform with the schema.prisma file and db datatypes are currently not supported
2. The dev command drops the db resulting in stuff not working because the preconfiguration is gone
3. It adds a prisma migrations table which is public and therefore added to the API

**Reply by kiwicopple** [reply]
That's useful to know @enricoschaaf 

> db datatypes are currently not supported

Do you know which datatypes are failing?

> adds a prisma migrations table which is public

This probably isn't an issue if you turn on RLS but I can understand that it looks untidy in the dashboard.

**Reply by enricoschaaf** [reply]
Oh I think I described it very poorly, it's not an issue with Supabase it's that prisma only has support for very basic datatypes, for example I wanted a date type but prisma currently only supports a datetime type, so I modified the schema to type date but now prisma tries to alter it to datetime every time I run a new migration which makes sense but is annoying. They will support native db types in the schema soon so that won't be an issue in the future

**Comment by maximousblk** [comment]
I was trying to get supabase working with prisma but I get this error when I run `prisma migrate dev`

![image](https://user-images.githubusercontent.com/28438021/121786421-4b82b400-cbdd-11eb-9a1c-5c44e77d6d76.png)

<details>
<summary>prisma schema</summary>

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  // DATABASE_URL="postgres://postgres:[PASSWORD]@db.[DATABASE].supabase.co:5432/postgres"
  url      = env("DATABASE_URL")
}

model Module {
  id          String   @id @default(dbgenerated("extensions.uuid_generate_v4()")) @db.Uuid
  name        String   @db.VarChar(64)
  fullname    String?
  description String?
  homepage    String?
  repository  String?
  issues      String?
  license     String   @default("UNKNOWN") @db.VarChar
  logo        String?
  verified    Boolean  @default(false)
  private     Boolean  @default(false)
  unlisted    Boolean  @default(false)
  createdAt   DateTime @default(now())
  updatedAt   DateTime @updatedAt

  authorId       String   @db.Uuid
  ContributorIds String[] @db.Uuid

  versions     Version[] @relation("rVersionToModule")
  author       User      @relation("rModuleToAuthor", fields: [authorId], references: [id])
  contributors User[]    @relation("rModuleToContributor", fields: [ContributorIds], references: [id])
}

model User {
  id        String   @id @default(dbgenerated("extensions.uuid_generate_v4()")) @db.Uuid
  username  String?  @db.VarChar(64)
  name      String?
  avatar    String?
  email     String?
  bio       String?
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  modules       Module[] @relation("rModuleToAuthor")
  contributions Module[] @relation("rModuleToContributor")
}

model Version {
  id        String   @id @default(dbgenerated("extensions.uuid_generate_v4()")) @db.Uuid
  tag       String
  createdAt DateTime @default(now())

  moduleId String @db.Uuid

  module Module @relation("rVersionToModule", fields: [moduleId], references: [id])
  files  File[] @relation("rFileToVersion")
}

model File {
  id        String   @id @default(dbgenerated("extensions.uuid_generate_v4()")) @db.Uuid
  path      String
  url       String
  mimeType  String   @db.VarChar(64)
  createdAt DateTime @default(now())

  versionId String @db.Uuid

  version Version @relation("rFileToVersion", fields: [versionId], references: [id])
}
```

</details>

Am I doing something wrong?
